iceoryx_binding_c/source/c_listener.cpp

```diff
@@ -203,3 +204,51 @@ void iox_listener_detach_client_event(iox_listener_t const self,
 
     self->detachEvent(*client, c2cpp::clientEvent(clientEvent));
 }
+
+iox_ListenerResult
+iox_listener_attach_service_discovery_event(iox_listener_t const self,
+                                            iox_service_discovery_t const serviceDiscovery,
+                                            const ENUM iox_ServiceDiscoveryEvent serviceDiscoveryEvent,
+                                            void (*callback)(iox_service_discovery_t))
+{
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(serviceDiscovery != nullptr);
+    iox::cxx::Expects(callback != nullptr);
+
+    auto result = self->attachEvent(*serviceDiscovery,
+                                    c2cpp::serviceDiscoveryEvent(serviceDiscoveryEvent),
+                                    NotificationCallback<ServiceDiscovery, internal::NoType_t>{callback, nullptr});
+
+    return (result.has_error()) ? cpp2c::listenerResult(result.get_error())
+                                : iox_ListenerResult::ListenerResult_SUCCESS;
+}
+
+iox_ListenerResult iox_listener_attach_service_discovery_event_with_context_data(
+    iox_listener_t const self,
+    iox_service_discovery_t const serviceDiscovery,
+    const ENUM iox_ServiceDiscoveryEvent serviceDiscoveryEvent,
+    void (*callback)(iox_service_discovery_t, void*),
+    void* const contextData)
+{
+    iox::cxx::Expects(self != nullptr);
+    iox::cxx::Expects(serviceDiscovery != nullptr);
+    iox::cxx::Expects(callback != nullptr);
+    iox::cxx::Expects(contextData != nullptr);
```

Comment:
We should remove this since a `nullptr` as `contextData` should be valid and not our problem. The user has to handle this.